dcos/packagemanager.py

```diff
@@ -308,12 +254,18 @@ def package_add(self, dcos_package):
         :return: Response to the package add request
         :rtype: Response
         """
-        request = "add"
-        headers = self._request_preferences().get(request).pop(0)
         with util.open_file(dcos_package, 'rb') as pkg:
-            headers['Content-MD5'] = util.md5_hash_file(pkg)
+            extra_headers = {'Content-MD5': util.md5_hash_file(pkg)}
         with util.open_file(dcos_package, 'rb') as data:
```

Comment:
Yes. the seek pointer is move all the way to the end of file (EOF). You have to seek to the beginning or re-open it like you are doing.